### PR TITLE
Bump bindgen to 0.31

### PR DESCRIPTION
**Pull request by Eijebong**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19043)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -3309,48 +3279,6 @@ dependencies = [
 ]
 
 [[package]]
-name = "syntex"
```
Comment (SimonSapin) [inline Cargo.lock]: 🎉🎉🎉

**Comment by SimonSapin** [comment]
r=me if Firefox builds with this change

**Comment by emilio** [comment]
I suspect it does not, but testing now to see

**Comment by Eijebong** [comment]
Currently blocked by:
- [ ] https://github.com/rust-lang-nursery/rust-bindgen/pull/1111
- [ ] https://github.com/rust-lang-nursery/rust-bindgen/pull/1112
- [ ] https://github.com/rust-lang-nursery/rust-bindgen/issues/1113

**Comment by emilio** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1412486 Has all the necessary changes for this to be feasible.

**Comment by Eijebong** [comment]
Closing in favour of #19072

**Pull request closed without merging**